lib/apiservers/engine/backends/container.go

```diff
@@ -1425,7 +1411,7 @@ func containerConfigFromContainerInfo(id string, info *models.ContainerInfo) *co
 
 	// Fill in information about the process
 	if info.ProcessConfig.ExecArgs != nil {
-		container.Env = info.ProcessConfig.ExecArgs // List of environment variable to set in the container
+		container.Env = info.ProcessConfig.Env // List of environment variable to set in the container
 	}
```

Comment:
That was actually the problem in #1594. It should have been using Env all along. This fixes it. :)